### DISCUSSION TITLE
Mailchimp group

### DISCUSSION THREAD

**Discussion by minalove** [solspace/craft-freeform#165] [Feature Requests]
Hi,

I'm trying to integrate the Mailchimp form and the form has a checkbox group which needs to be linked to a Mailchimp Group. I couldn't find a mapping function in the Freeform panel. Is Freeform support the mapping for Mailchimp Group?

Many thanks!

**Comment by kjmartens** [comment]
Unfortunately, this feature is not yet available for the MailChimp integration for Freeform. It's on our list for future consideration though. :)

**Comment by kjmartens** [comment]
This has now been implemented in Freeform 3.13.16+ and 4.0.0-beta.17+. It will just allow the mapping of a single group at a time, however. 🙂 

**UPDATE July 5, 2023**
Added support for mapping to multiple groups/interests in Freeform 4.1.4+

**Comment by shpfive** [comment]
Is there a plan to integrate the option for multiple groups?

**Reply by shpfive** [reply]
I forked the project and was able to implement multiple groups the same way as the tags are handled.
I split the value and loop over them to get the matching id and set them in the member data.
it was easier than expected.

```php
if (preg_match('/interests___interests/', $key)) {
  $interestsSelection = explode(',', $value);
  $interestsSelection = array_map('trim', $interestsSelection);
  $interestsSelection = array_filter($interestsSelection);

  foreach ($interestsSelection as $interest){
    $interestId = $this->findInterestIdFromName($interest, $listId);
    if ($interestId) {
        $interests[$interestId] = true;
    }
  }

  unset($mappedValues[$key]);
}
``` 

Any interest in help from my side?

**Reply by kjmartens** [reply]
Thanks for this @shpfive. We'll take a look at this next week and add support for it if it's as easy as suggested without any negative side effects, etc. 🙂 

If you wish to send a PR, you're also welcome to do that as well. 🙂

**Comment by kjmartens** [comment]
We've now added support for mapping to multiple groups/interests in Freeform 4.1.4+.

Thanks for the code suggestion @shpfive 🙂

**Reply by shpfive** [reply]
Thank you for implementing the multi group option.
Is it possible to also add that to v3? I have some older projects where that would be helpfull.

im always happy to help :)

**Reply by kjmartens** [reply]
Sure, we'll add it for v3 as well. I've queued it up and we'll likely have it available within the next week or so. 🙂

**Reply by kjmartens** [reply]
@shpfive We've added this in v3.13.31 now as well. 🙂

**Reply by shpfive** [reply]
Thank you :)
